src/track/track.cpp

```diff
@@ -891,30 +904,34 @@ void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
 }
 
 Track::ImportStatus Track::importBeats(
-        mixxx::BeatsImporterPointer pBeatsImporter) {
+        mixxx::BeatsImporterPointer pBeatsImporter,
+        bool bpmLocked) {
     QMutexLocker lock(&m_qMutex);
     VERIFY_OR_DEBUG_ASSERT(pBeatsImporter) {
         return ImportStatus::Complete;
     }
     DEBUG_ASSERT(!m_pBeatsImporterPending);
     m_pBeatsImporterPending = pBeatsImporter;
     if (m_pBeatsImporterPending->isEmpty()) {
-        // Just return the current import status without clearing any
-        // existing cue points.
         m_pBeatsImporterPending.reset();
         return ImportStatus::Complete;
     } else if (m_record.hasStreamInfoFromSource()) {
-        // Replace existing cue points with imported cue
-        // points immediately
-        importPendingBeatsMarkDirtyAndUnlock(&lock);
+        // Replace existing beats with imported beats immediately
+        importPendingBeatsMarkDirtyAndUnlock(&lock, bpmLocked);
         return ImportStatus::Complete;
     } else {
         kLogger.debug()
                 << "Import of beats is pending until the actual sample rate becomes available";
-        // Clear all existing cue points, that are supposed
-        // to be replaced with the imported cue points soon.
-        setBeatsMarkDirtyAndUnlock(&lock, nullptr);
-        return ImportStatus::Pending;
+        // Clear all existing beats, that are supposed
+        // to be replaced with the imported beats soon.
+        if (trySetBeatsMarkDirtyAndUnlock(&lock,
+                    nullptr,
+                    bpmLocked ? SetBeatsMode::SetBpmLock
+                              : SetBeatsMode::KeepBpmLock)) {
```

Comment:
What does this mean? I don't understand why `bpmLocked` maps to either `SetBpmLock` or `KeepBpmLock`. The variable sounds like Bpm Lock is either of or off.

Reply:
KeepBpmLock keeps the bpm lock unchanged. SetBpmLock locks the just set bpm value.

The stored bpm lock value does not unlock the bpm, because the user has intentional locked the bpm before to protect it from any kind of changes.

Do you have an idea for improved names?

Reply:
The confusion is not about the enum names but how the boolean `bpmLocked` is interpreted. This is not obvious and I don't understand the mapping.

Reply:
Why do we keep the bpm lock if `bpmLocked = false`? We should use the enum consistently and not keep this confusing bool here. The name doesn't match what it actually does.

Reply:
We keep the lock if the bpm lock is locked, because only the user is responsible for unlocking the lock.

Reply:
I am happy to change the enum names or add s comment. Can you propose something?

Reply:
Pass `SetBeatsMode mode` as a parameter instead of `bool bpmLocked` to avoid this impedance mismatch.

Reply:
But obviously this doesn't work. Nevertheless we lose information here when discarding bpmLocked = false (to reset the lock) and instead keep the existing state. The function behaves differently depending on some internal circumstances and that makes me nervous.

Reply:
Isn't that the nature of bpm lock to discard any changes. The function now has the try.. prefix.

Reply:
If I pass bpmLocked = false I would expect that the lock is reset by this function in any case.

Reply:
The other function has a "try" prefix but not this function.